### PR TITLE
feat(favorite): #MAG-223 add sort by fav in card's collection

### PR DESCRIPTION
**Pull request by Sofianebernoussi**
## Describe your changes
Add a feature allowing the user to order by favorite in card collection.
## Checklist tests
Use the toggle button in card collection to order by favorite, to see them in priority. You can do the same operation when you group by board.
## Issue ticket number and link
[MAG-223](https://jira.support-ent.fr/browse/MAG-223)
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/cgi/magneto/service/impl/DefaultCardService.java [hunk 1]
```diff
@@ -700,8 +700,23 @@ private JsonObject getAllCardsQuery(UserInfos user, String boardId, Integer page
         query
                 .match(new JsonObject().put(String.format("%s.%s", Field.RESULT, Field.DELETED), false))
                 .sort(Field.PARENTID, 1)
-                .group(groupField, externalGroupField)
-                .sort(String.format("%s.%s", Field.RESULT, Field.MODIFICATIONDATE), -1);
+                .group(groupField, externalGroupField);
```
Comment (Sofianebernoussi) [inline src/main/java/fr/cgi/magneto/service/impl/DefaultCardService.java:703]: utilise le champ d'alex

File: src/main/resources/i18n/fr.json [hunk 2]
```diff
@@ -163,6 +163,7 @@
   "magneto.cards.collection.shared": "Dans les tableaux partagés",
   "magneto.cards.collection.public": "Dans les tableaux de la plateforme",
   "magneto.cards.collection.board.view": "Ranger par tableau",
+  "magneto.cards.collection.favorite.view": "Trier par like",
```
Comment (Sofianebernoussi) [inline src/main/resources/i18n/fr.json:166]: autre langue

File: src/main/resources/public/ts/directives/card-collection/card-collection.directive.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ import {CardsFilter} from "../../models/cards-filter.model";
 import {Object} from "core-js";
 import {InfiniteScrollService} from "../../shared/services";
 import {COLLECTION_NAVBAR_VIEWS} from "../../core/enums/collection-navbar.enum";
+import {RESOURCE_TYPE} from "../../core/enums/resource-type.enum";
```
Comment (Sofianebernoussi) [inline src/main/resources/public/ts/directives/card-collection/card-collection.directive.ts:12]: a delete

File: src/main/resources/public/ts/directives/card-collection/card-collection.directive.ts [hunk 4]
```diff
@@ -141,6 +146,23 @@ class Controller implements IViewModel {
         safeApply(this.$scope);
     }
 
+    changeOrder = async (field: string): Promise<void> => {
+        switch (field) {
+            case "favorite":
```
Comment (Sofianebernoussi) [inline src/main/resources/public/ts/directives/card-collection/card-collection.directive.ts:151]: magic

**Review by flmartineau — APPROVED** [review]

**Pull request merged**